packages/ai-genkit/src/retrievers/fga-retriever.ts

```diff
@@ -65,22 +64,30 @@ export class FGARetriever {
   /**
    * Creates a new FGARetriever instance for filtering documents based on custom query logic.
    *
-   * @param args - The properties used to configure the FGARetriever. Query builder and the base retriever.
+   * @param args.buildQuery - A function that checks the FGARetriever query.
+   * @param args.retriever - An optional GenKit retriever.
+   * @param args.ai - A GenKit Instance.
    * @param fgaClient - An optional OpenFgaClient instance.
    * @returns A RetrieverAction instance.
    */
   static create<CustomOptions extends z.ZodTypeAny = z.ZodTypeAny>(
+    ai: Genkit,
     { buildQuery, retriever }: FGARetrieverProps<CustomOptions>,
     fgaClient?: OpenFgaClient
   ) {
     const fga = new FGARetriever({ buildQuery }, fgaClient);
 
-    return defineRetriever(
+    const fgaRetrieverOptionsSchema = CommonRetrieverOptionsSchema.extend({
+      preRerankK: z.number().max(1000), // TODO: check if this is the correct max value
```

Comment:
@jcenturion not usre what is the right number to use here, do you know this?

Reply:
No, maybe it should configurable by the dev?

Reply:
ya that could be an option with a default when not set

Reply:
done